cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/workbenchcontrol/handler/AdaptTemplatesHandler.java

```diff
@@ -0,0 +1,191 @@
+     * structure at /main/CobiGen_Templates location
+     * @param fileName
+     *            Name of source jar file downloaded
+     */
+    private void processJar(String fileName) {
+
+        String jarPath = ws.toPortableString() + ResourceConstants.DOWNLOADED_JAR_FOLDER + "/" + fileName;
+        String pathForCobigenTemplates = ws.toPortableString() + ResourceConstants.COBIGEN_TEMPLATES_FOLDER;
+        FileSystem fileSystem = FileSystems.getDefault();
+        Path cobigenFolderPath = fileSystem.getPath(pathForCobigenTemplates);
+        Path configFolder = fileSystem.getPath(ws.toPortableString() + "/main/CobiGen_Templates/src/main/templates");
+        ContextConfiguration contextConfiguration = new ContextConfiguration(configFolder);
+        List<String> templateNames = new ArrayList<>();
+        for (Trigger t : contextConfiguration.getTriggers()) {
+            templateNames.add(t.getTemplateFolder());
+        }
+
+        try (ZipFile file = new ZipFile(jarPath)) {
+            deleteDirectoryStream(configFolder);
+            Enumeration<? extends ZipEntry> entries = file.entries();
+            if (Files.notExists(cobigenFolderPath)) {
+                Files.createDirectory(cobigenFolderPath);
+            }
+            while (entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+                Path saveForFileCreationPath = fileSystem.getPath(cobigenFolderPath + File.separator + entry.getName());
+                if (templateNames.parallelStream().anyMatch(entry.getName()::contains)
+                    || entry.getName().contains("context.xml")) {
+                    saveForFileCreationPath = fileSystem.getPath(cobigenFolderPath + File.separator + "src"
+                        + File.separator + "main" + File.separator + "templates" + File.separator + entry.getName());
+                } else if (entry.getName().contains("com/")) {
+                    saveForFileCreationPath = fileSystem.getPath(cobigenFolderPath + File.separator + "src"
+                        + File.separator + "main" + File.separator + "java" + File.separator + entry.getName());
+                }
+                if (entry.isDirectory()) {
+                    Files.createDirectories(saveForFileCreationPath);
+                } else {
+                    Files.deleteIfExists(saveForFileCreationPath);
+                    InputStream is = file.getInputStream(entry);
+                    BufferedInputStream bis = new BufferedInputStream(is);
+                    Files.createFile(saveForFileCreationPath);
+                    FileOutputStream fileOutput = new FileOutputStream(saveForFileCreationPath.toString());
+                    while (bis.available() > 0) {
+                        fileOutput.write(bis.read());
+                    }
+                    bis.close();
+                    fileOutput.close();
```

Comment:
Use try-with-resource to manage streams to prevent from resource leaks and open handles